### PR TITLE
x fmt: Don't print all modified files if there's more than 10

### LINKED ISSUE #106920: `x.py fmt` is extremely noisy
Today I synchronized my local version of the rust repo, made a few changes and formatted. 

This prints:
```
Building rustbuild
formatting modified file compiler/rustc_abi/src/lib.rs
formatting modified file compiler/rustc_ast/src/ast.rs
formatting modified file compiler/rustc_ast/src/ast_traits.rs
formatting modified file compiler/rustc_ast/src/attr/mod.rs

/* 35k lines omitted for brevity */

formatting modified file tests/ui/zero-sized/zero-sized-binary-heap-push.rs
formatting modified file tests/ui/zero-sized/zero-sized-btreemap-insert.rs
formatting modified file tests/ui/zero-sized/zero-sized-linkedlist-push.rs
formatting modified file tests/ui/zero-sized/zero-sized-tuple-struct.rs
Build completed successfully in 0:00:05
```

### PR DESCRIPTION
**Pull request by jyn514**
This avoids spam for dozens of modified files.

Fixes https://github.com/rust-lang/rust/issues/106920.

### CODE REVIEW

**Comment by albertlarsan68** [comment]
r? @albertlarsan68
Thanks for the PR!
@bors rollup
r=me whe CI green

**Comment by jyn514** [comment]
@bors r=albertlarsan68 

Thanks!

File: src/bootstrap/format.rs [hunk 1]
```diff
@@ -159,8 +159,14 @@ pub fn format(build: &Builder<'_>, check: bool, paths: &[PathBuf]) {
             if !check && paths.is_empty() {
                 match get_modified_rs_files(build) {
                     Ok(Some(files)) => {
+                        if files.len() <= 10 {
+                            for file in &files {
+                                println!("formatting modified file {file}");
+                            }
+                        } else {
+                            println!("formatting only modified files");
```
Comment (Mark-Simulacrum) [inline src/bootstrap/format.rs:167]: Can we add the count? It'll help with detecting if our modified detection is wrong (e.g. if the number is in thousands).

**Comment by albertlarsan68** [comment]
@bors R-A
Please address Mark's comment

**Comment by albertlarsan68** [comment]
@bors r-
Please address Mark's comment

**Pull request merged**